ui/src/components/UI/SelectList.tsx

```diff
@@ -0,0 +1,97 @@
+import { useMemo } from "react";
+import Checkbox from "./Checkbox"
+import Radio from "./Radio";
+import './style/SelectList.sass';
+import NoDataMessage from "./NoDataMessage";
+
+export type ValuesListInput = {
+    key: string;
+    value: string;
+}[]
+export interface Props {
+    items;
+    tableName:string;
+    checkedValues?:string[];
+    multiSelect:boolean;
+    searchValue?:string;
+    setCheckedValues: (newValues)=> void;
+    tabelClassName
+}
+
+const SelectList: React.FC<Props> = ({items ,tableName,checkedValues=[],multiSelect=true,searchValue="",setCheckedValues,tabelClassName}) => {
+
+    const noItemsMessage = "No items to show";
+ 
+    const filteredValues = useMemo(() => {
+        return items.filter((listValue) => listValue?.value?.includes(searchValue));
+    },[items, searchValue])
+
+    const toggleValue = (checkedKey) => {
+        if (!multiSelect){
+            const newCheckedValues = [];
+            newCheckedValues.push(checkedKey);
+            setCheckedValues(newCheckedValues);
+        }
+        else {
+            const newCheckedValues = [...checkedValues];
+            let index = newCheckedValues.indexOf(checkedKey);
+            if(index > -1) 
+                newCheckedValues.splice(index,1);
+            else 
+                newCheckedValues.push(checkedKey);   
+            setCheckedValues(newCheckedValues);
+        }
+    }
+
+    const toggleAll = () => {
+        const newCheckedValues = [...checkedValues];
+        if(newCheckedValues.length === items.length) setCheckedValues([]);
+        else {
+            items.forEach((obj) => {
+            if(!newCheckedValues.includes(obj.key))
+                newCheckedValues.push(obj.key);
+            })
+            setCheckedValues(newCheckedValues);
```

Comment:
can be replaced with:
else {
     setCheckedValues(items.map(item => item.key));
}